src/Pate/Equivalence.hs

```diff
@@ -326,11 +327,12 @@ memDomPost ::
   MemRegionEquality sym arch ->
   SimOutput sym arch PBi.Original ->
   SimOutput sym arch PBi.Patched ->
-  PEM.MemoryDomain sym arch ->
+  PEM.MemoryDomain sym arch {- ^ pre-domain for this slice -} ->
+  PEM.MemoryDomain sym arch {- ^ target post-domain -} ->
```

Comment:
The haddock for this function refers to "the given 'PEM.MemPred'". It now takes two `PEM.MemoryDomain` instead. Can you update the comment to reflect that? This change seems slightly subtle, so we should capture what the semantics are here (and it is now important that it gets both a pre and post).

Reply:
Added some comments explaining what each given MemoryDomain is used for